src/components/tl/cuda/alltoallv/alltoallv_ce.c

```diff
@@ -0,0 +1,402 @@
+    default:
+        ucc_assert(task->alltoallv_ce.stage == ALLTOALL_CE_STAGE_BAR);
+        break;
+    }
+    task->super.super.status =
+        ucc_tl_cuda_shm_barrier_test(UCC_TL_TEAM_RANK(team), task->bar);
+    if (task->super.super.status == UCC_OK) {
+        ucc_tl_cuda_put_sync(task);
+    }
+    return task->super.super.status;
+}
+
+ucc_status_t ucc_tl_cuda_alltoallv_ce_start(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_cuda_task_t *task = ucc_derived_of(coll_task, ucc_tl_cuda_task_t);
+    ucc_tl_cuda_team_t *team = TASK_TEAM(task);
+
+    if (task->alltoallv_ce.stage != ALLTOALL_CE_STAGE_POST_COPIES) {
+        task->alltoallv_ce.stage = ALLTOALL_CE_STAGE_SYNC;
+    }
+
+    ucc_tl_cuda_alltoallv_ce_progress(coll_task);
+    if (task->super.super.status == UCC_INPROGRESS) {
+        ucc_progress_enqueue(UCC_TL_CORE_CTX(team)->pq, &task->super);
+        return UCC_OK;
+    }
+    return ucc_task_complete(coll_task);
+}
+
+ucc_status_t
+ucc_tl_cuda_alltoallv_ce_triggered_post_setup(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_cuda_task_t *task = ucc_derived_of(coll_task, ucc_tl_cuda_task_t);
+    ucc_status_t        status;
+
+    do {
+        status = ucc_tl_cuda_get_sync(task);
+    } while (status == UCC_INPROGRESS);
+    if (ucc_unlikely(status != UCC_OK)) {
+        return status;
+    }
+
+    status = ucc_tl_cuda_alltoallv_setup_start(task);
+    if (ucc_unlikely(status != UCC_OK)) {
+        ucc_tl_cuda_put_sync(task);
+        return status;
+    }
+
+    do {
+        status = ucc_tl_cuda_alltoallv_setup_test(task);
+    } while (status == UCC_INPROGRESS);
+    if (ucc_unlikely(status != UCC_OK)) {
+        ucc_tl_cuda_put_sync(task);
+        return status;
+    }
+    task->alltoallv_ce.stage = ALLTOALL_CE_STAGE_POST_COPIES;
+
+    return UCC_OK;
+}
+
+//NOLINTNEXTLINE
+size_t ucc_tl_cuda_alltoallv_get_count(const ucc_tl_cuda_task_t *task,
+                                       ucc_count_t *cnts, ucc_rank_t block)
+{
+    const ucc_coll_args_t *args = &TASK_ARGS(task);
+    return ucc_coll_args_get_count(args, cnts, block);
+}
+
+size_t ucc_tl_cuda_alltoallv_get_offset(const ucc_tl_cuda_task_t *task,
+                                        size_t *displ, ucc_rank_t block)
+{
+    const ucc_coll_args_t *args = &TASK_ARGS(task);
+    return ucc_coll_args_get_displacement(args, displ, block);
+}
+
+ucc_status_t ucc_tl_cuda_alltoallv_ce_init(ucc_tl_cuda_task_t *task)
+{
+    ucc_tl_cuda_team_t *team = TASK_TEAM(task);
+    ucc_coll_args_t    *args = &TASK_ARGS(task);
+    ucc_status_t        status;
+    size_t              data_len;
+
+    if (!UCC_COLL_ARGS_CONTIG_BUFFER(args)) {
+        tl_error(UCC_TL_TEAM_LIB(team), "Do not support non-contiguous buffer");
+        return UCC_ERR_NOT_SUPPORTED;
+    }
+
+    status =
+        ucc_ec_create_event(&task->alltoallv_ce.copy_done, UCC_EE_CUDA_STREAM);
+    if (ucc_unlikely(status != UCC_OK)) {
+        return status;
+    }
+
```

Comment:
need to add task->super.flags |= UCC_COLL_TASK_FLAG_EXECUTOR to tell upper layer to create executor for regular post